tools_for_todoist/models/event.py

```diff
@@ -121,15 +126,46 @@ def _last_occurrence(self):
         if not is_allday(start):
             last_occurence = last_occurence.astimezone(start.tzinfo)
         return last_occurence.date() if is_allday(start) else last_occurence
+    
+    def _find_next_occurence(self, rrule_instances):
+        first_exception_start = min(
+            (
+                start
+                for start in (
+                    x._get_start() 
+                    for x in self.exceptions.values()
+                    if not x.is_cancelled
+                )
+                if start >= (datetime.now() if is_allday(start) else datetime.now(UTC))
+            ),
+            default=None,
+        )
+        invalid_starts = [
```

Comment:
these aren't invalid, you can have an exception to an event but the start date could be the same as before and you shouldn't skip it.